### DISCUSSION TITLE
Security Question

### DISCUSSION THREAD

**Discussion by goochusmc** [squidfunk/mkdocs-material#4399] [Q&A] [answered]
I am wondering if there are any known security issues with the theme? I was trying to switch to this theme for project documentation at work and when I tried to pip install it was blocked. After some experimentation I contacted my IT department and was told that the "real-time security processing and anti-malware detection flagged it as containing malicious content.

**Comment by squidfunk** [comment, marked as answer]
Thanks for asking. I'm not aware of any open security issues. We would need to know the reason why the installation was blocked to investigate. Also, [Snyk regularly scans Material for MkDocs](https://snyk.io/advisor/python/mkdocs-material) and doesn't list any security vulnerabilities, [actually never](https://security.snyk.io/package/pip/mkdocs-material).

**Reply by goochusmc** [reply]
I don't have any more information than what I provided above. I suspect that my IT department does not either. They are likely relying entirely on whatever the automated detection system says without understanding at all what it means.

**Reply by squidfunk** [reply]
Well, I'm afraid I'm not able to help then. We don't ship any malware or malicious content. Material for MkDocs is used by hundreds or even thousands of companies without any security issues. In fact, I'm hearing this for the first time. It might be a good idea to talk to the vendor supplying the malware detection software.

**Reply by goochusmc** [reply]
I understand and am not suggesting I have given you enough info to do anything. I was just looking for some info to maybe use to push back on my IT department a bit. I have asked if they can provide any more detailed info and will let you know if they provide anything that might be actionable for you.

**Reply by goochusmc** [reply]
I was able to get the report that our IP department gets on this. The link below should allow you to view it. This is a [Forcepoint](https://www.forcepoint.com/) report so not something generated by my organization. If I am reading this correctly, which I might not be, it looks like the download IP is being flagged as "compromised to serve malicious content." I can't really get anything further from my IT department as [Forcepoint](https://www.forcepoint.com/) are the ones making this determination.

https://csi.forcepoint.com/Report/Index/de11cabb-708f-4ac6-9ef1-af23007c0a6d

Edit:

My IT people have a different interpretation of this laguage. I got the following from them.

"My interpretation is that they are fine with the site in general but are flagging the specific download. 

-  In the Executive Summary -> Classification section on the right it indicates the site itself is categorized as IT and the real-time content analysis for the site is also IT.  If they had general concerns about the site the static and/or real-time categorization would be something like ‘Malicious Web Sites’ or something besides IT.
-  In the Executive Summary -> Threat Severity section and the Assessment Overview they indicate the concern is with the specific URL / file being downloaded.  I think the ‘compromised to serve malicious content’ is generic language they use, with supply chain attacks affecting python libraries the hosting site doesn’t need to be compromised the package payload just needs to be. 

It would be nice if there was more detail in the Security Overview -> Real-time security identification section, but that could be due to propriety detection methods / signatures.  I would not expect any malicious link detection's on the package as it’s a zip file, not a web page."

**Reply by squidfunk** [reply]
Thanks for coming back. It looks like the IP is belonging to `pypi.python.org`:

```
$ nslookup pypi.python.org

Server:		192.168.8.1
Address:	192.168.8.1#53

Non-authoritative answer:
pypi.python.org	canonical name = dualstack.python.map.fastly.net.
Name:	dualstack.python.map.fastly.net
Address: 151.101.112.223
```

This means, that the vendor has flagged (some of) the download servers of PyPI as a security threat. There's nothing that we can do here. You could try and [install via git](https://squidfunk.github.io/mkdocs-material/getting-started/#with-git) or [download the zip](https://github.com/squidfunk/mkdocs-material/releases) and install it manually, but that might be flagged as well.

**Reply by goochusmc** [reply]
That confuses me. I am fine downloading virtually any other package from pypi.python.org without any issue. It seems to just be this particular package. 

I did try downloading the whl file directly but that gets flagged too.

**Reply by squidfunk** [reply]
That is very odd. As said, the package doesn't contain any executables or malicious content. If I can do anything to help on our front, I'm happy to do so, but without more information, it's unclear what we would need to do to get it unflagged.

**Reply by squidfunk** [reply]
As an alternative, you might try running the official [Docker image](https://hub.docker.com/r/squidfunk/mkdocs-material/).

**Reply by goochusmc** [reply]
There is probably not any more information I can give you from my organization as it is not my organization flagging this. I just thought you may want to contact [Forcepoint](https://www.forcepoint.com/) to understand why they are flagging your stuff. My organization is just relying on their checks and they are apparently detecting something with this so it would seem any organization using [Forcepoint](https://www.forcepoint.com/) will have this same issue.

**Reply by squidfunk** [reply]
I'm afraid I don't have the time or resources to contact Forcepoint or other vendors. Any help is appreciated.

**Reply by goochusmc** [reply]
When I get some time I will try to reach out to them for you.

**Reply by squidfunk** [reply]
Thanks!
